### ISSUE TITLE
Preference setting for disabling *all* keyboard shortcuts

### ISSUE THREAD

**Issue by azonenberg** [mastodon/mastodon#21688]
### Pitch

There should be a preference setting, most likely under accessibility, for disabling all keyboard shortcuts in the web UI.

### Motivation

Some users have trouble making precise mouse inputs and/or occasionally make unwanted mouse inputs. This could be due to a hand dragging on a touchpad, a mobility-related disability making a mouse hard to us, a pet or child climbing on their desk, etc.

If a user begins typing a toot thinking they're in the "compose toot" box but focus is in fact somewhere else, it's easy to accidentally perform a bunch of unfollows, boosts, or other unwanted actions. Pressing enter/space while typing a toot in the wrong place can even result in unintentionally approving a confirmation dialog.

With this preference setting active, all keyboard input outside of the "compose toot" box should be ignored.

**Comment by lapcat** [comment]
Agreed. Just resting my hands on the keyboard, I'll occasionally make a stray keystroke. I don't want that to accidentally cause an action. Back when I used Twitter, I had to write a web browser extension to stop it from getting key presses, because it also has single key shortcuts.

It almost works to block every keydown except "Escape" on Mastodon, but for some reason the search field relies on the "Enter" key instead of using a <form>.
